### PR TITLE
Match CheckBagHasSpace

### PR DESCRIPTION
**Pull request by PokeCodec**
This leaves one final function left to decompile and fix (with HandleAction_UseItem being the culprit).

### CODE REVIEW

**Review by PokeCodec — COMMENTED** [review]
The switch to ``if ((*(gBattleStruct->AI_itemFlags + gBattlerAttacker / 2) & 1)
            && (*(gBattleStruct->AI_itemFlags + gBattlerAttacker / 2) & 0x3E))`` SHOULD have caused a nonmatching, but it did not.

Thought this was too good of a quirk to not put in this PR.

**Comment by PokeCodec** [comment]
I'm going to say this PR is ready for review.

File: src/m4a_1.s [hunk 1]
```diff
@@ -20,7 +20,7 @@ __umul3232H32:
 	thumb_func_start SoundMain
 SoundMain:
 	ldr r0, lt_SOUND_INFO_PTR
-	ldr r0, [r0]
+	ldr r0, [r0] @ Set the SoundInfo pointer to sound area
```
Comment (PikalaxALT) [inline src/m4a_1.s:23]: Comment incorrect

**Review by PikalaxALT — CHANGES_REQUESTED** [review]

**Comment by PokeCodec** [comment]
Fixed

**Review by PikalaxALT — APPROVED** [review]

**Pull request merged**